resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala

```diff
@@ -87,11 +89,37 @@ class KubernetesConfSuite extends SparkFunSuite {
       APP_ID,
       None,
       MAIN_CLASS,
-      APP_ARGS)
+      APP_ARGS,
+      None)
     assert(kubernetesConfWithoutMainJar.sparkConf.get("spark.jars").split(",")
       === Array("local:///opt/spark/jar1.jar"))
+    assert(kubernetesConfWithoutMainJar.sparkConf.get(MEMORY_OVERHEAD_FACTOR) === 0.1)
   }
 
+  test("Creating driver conf with a python primary file") {
```

Comment:
Would like also see a unit test for with a PyFile and an overriden memory overhead.

Reply:
Defaults are checked on 96 and 117. (But I need to ensure that it is possible to override as well. Will add)

Reply:
Just a follow up we should have a test for with Python and overriding MEMORY_OVERHEAD_FACTOR (e.g. test to make sure that setIfMissing since we had it the other way earlier in the PR).